Files/Views/SettingsPages/Preferences.xaml.cs

```diff
@@ -1,12 +1,29 @@
-﻿using Windows.UI.Xaml.Controls;
+﻿using Files.Filesystem;
+using Windows.UI.Xaml.Controls;
 
 namespace Files.SettingsPages
 {
     public sealed partial class Preferences : Page
     {
+        public static LibraryManager LibraryManager { get; private set; }
+
         public Preferences()
         {
+            LibraryManager ??= new LibraryManager();
+
             InitializeComponent();
         }
+
+        private async void ShowLibrarySection_Toggled(object sender, Windows.UI.Xaml.RoutedEventArgs e)
+        {
+            if (App.AppSettings.ShowLibrarySection)
+            {
+                await LibraryManager.EnumerateDrivesAsync();
+            }
+            else
+            {
+                await LibraryManager.RemoveEnumerateDrivesAsync();
+            }
+        }
```

Comment:
Can you move this code to the settings view model for preferences?

Reply:
yes, please check on my last comit

Reply:
@gave92 Isn't this a little redundant since it's already bound to the property?